commit 8c0e30e4e8ca53822bda6b7e24c87ca9a9bdfd4a (refs/changes/41/441/1)
Author: David Pacheco <dap@joyent.com>
Date:   2016-09-08T17:51:38-07:00 (3 years, 1 month ago)
    
    MANTA-2804 lackey crash: CatStreams ended after previously aborted
    MANTA-2983 marlin tst.adnscache.js fails
    MANTA-2984 marlin should use semver for node-manta dependency

diff --git a/agent/lib/agent/lackey.js b/agent/lib/agent/lackey.js
index 356f7ab..cdea975 100644
--- a/agent/lib/agent/lackey.js
+++ b/agent/lib/agent/lackey.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright (c) 2014, Joyent, Inc.
+ * Copyright (c) 2016, Joyent, Inc.
  */
 
 /*
@@ -568,7 +568,8 @@ function mazDoneBatch(batch)
 			process.exit(1);
 		}
 
-		if (batch['taskInputDone'] || mazCurrentTask === undefined)
+		if (batch['taskInputDone'] || mazCurrentTask === undefined ||
+		    mazExecutorResult !== undefined)
 			return;
 
 		mazParentRequest('GET /my/jobs/task/task?wait=true', 200,
@@ -581,6 +582,10 @@ function mazDoneBatch(batch)
 				process.exit(1);
 			}
 
+			if (mazExecutorResult !== undefined) {
+				return;
+			}
+
 			mazLog.info('next batch', task);
 			mod_assert.equal(mazCurrentTask['taskId'],
 			    task['taskId']);
diff --git a/dev/package.json b/dev/package.json
index 61561d3..17bcaea 100644
--- a/dev/package.json
+++ b/dev/package.json
@@ -28,7 +28,7 @@
 		"libmanta": "git+ssh://git@github.com:joyent/node-libmanta.git#master",
 		"lstream": "0.0.3",
 		"mahi": "git+ssh://git@github.com:joyent/node-mahi.git#master",
-		"manta": "git+ssh://git@github.com:joyent/node-manta.git#master",
+		"manta": "^3.0.0",
 		"manta-compute-bin": "git+ssh://git@github.com:joyent/manta-compute-bin.git#master",
 		"memorystream": "0.2.0",
 		"mkdirp": "0.3.1",
diff --git a/dev/test/util/tst.adnscache.js b/dev/test/util/tst.adnscache.js
index f59adde..d51a116 100644
--- a/dev/test/util/tst.adnscache.js
+++ b/dev/test/util/tst.adnscache.js
@@ -70,13 +70,13 @@ function setup(_, next)
 	});
 
 	mod_assert.throws(function () {
-		cache.lookupv4('www.joyent.com');
-	}, '"www.joyent.com" is not known to this cache');
+		cache.lookupv4('us-east.manta.joyent.com');
+	}, '"us-east.manta.joyent.com" is not known to this cache');
 
-	cache.add('www.joyent.com');
-	mod_assert.ok(cache.lookupv4('www.joyent.com') === null);
+	cache.add('us-east.manta.joyent.com');
+	mod_assert.ok(cache.lookupv4('us-east.manta.joyent.com') === null);
 	cache.update();
-	mod_assert.ok(cache.lookupv4('www.joyent.com') === null);
+	mod_assert.ok(cache.lookupv4('us-east.manta.joyent.com') === null);
 
 	cache.add('127.0.0.1');
 	mod_assert.equal(cache.lookupv4('127.0.0.1'), '127.0.0.1');
@@ -90,7 +90,7 @@ function checkResolved(_, next)
 	 * Check that we've resolved the IP, and an immediate "update" doesn't
 	 * bother making a new request.
 	 */
-	ip = cache.lookupv4('www.joyent.com');
+	ip = cache.lookupv4('us-east.manta.joyent.com');
 	mod_assert.ok(looksLikeIp4(ip));
 	mod_assert.equal(0, cache.update());
 	setTimeout(function () { next(); }, 2500);
@@ -106,7 +106,7 @@ function forceUpdate(_, next)
 	advanced = false;
 	mod_assert.equal(1, cache.update());
 	mod_assert.equal(0, cache.update());
-	ip = cache.lookupv4('www.joyent.com');
+	ip = cache.lookupv4('us-east.manta.joyent.com');
 	mod_assert.ok(looksLikeIp4(ip));
 }
 
@@ -117,7 +117,7 @@ function checkResolvedAgain(_, next)
 	 * the grace period.
 	 */
 	log.info('checkResolved');
-	ip = cache.lookupv4('www.joyent.com');
+	ip = cache.lookupv4('us-east.manta.joyent.com');
 	mod_assert.ok(looksLikeIp4(ip));
 	mod_assert.equal(0, cache.update());
 	setTimeout(function () { next(); }, 4000);
@@ -126,7 +126,7 @@ function checkResolvedAgain(_, next)
 function checkGrace(_, next)
 {
 	log.info('checkGrace');
-	ip = cache.lookupv4('www.joyent.com');
+	ip = cache.lookupv4('us-east.manta.joyent.com');
 	mod_assert.ok(looksLikeIp4(ip));
 	setTimeout(function () { next(); }, 2000);
 }
@@ -134,7 +134,7 @@ function checkGrace(_, next)
 function checkGraceExpired(_, next)
 {
 	log.info('checkGraceExpired');
-	ip = cache.lookupv4('www.joyent.com');
+	ip = cache.lookupv4('us-east.manta.joyent.com');
 	mod_assert.ok(ip === null);
 	next();
 }
